lib/utils/clamp_path.py

```diff
@@ -0,0 +1,121 @@
+from shapely.geometry import LineString, Point as ShapelyPoint, MultiPolygon
+from shapely.prepared import prep
+from .geometry import Point, ensure_multi_line_string
+
+
+def path_to_segments(path):
+    """Convert a path of Points into a list of segments as LineStrings"""
+    for start, end in zip(path[:-1], path[1:]):
+        yield LineString((start, end))
+
+
+def segments_to_path(segments):
+    """Convert a list of contiguous LineStrings into a list of Points."""
+    coords = [segments[0].coords[0]]
+
+    for segment in segments:
+        coords.extend(segment.coords[1:])
+
+    return [Point(x, y) for x, y in coords]
+
+
+def fix_starting_point(border_pieces):
+    """Reconnect the starting point of a polygon border's pieces.
+
+    When splitting a polygon border with two lines, we want to get two
+    pieces.  However, that's not quite how Shapely works.  The outline
+    of the polygon is a LinearRing that starts and ends at the same place,
+    but Shapely still knows where that starting point is and splits there
+    too.
+
+    We don't want that third piece, so we'll reconnect the segments that
+    touch the starting point.
+    """
+
+    if len(border_pieces) == 3:
```

Comment:
better use border_pieces.geoms since shapely says it's deprecated to use it like that directly.